drivers/hwinfo/hwinfo_weak_impl.c

```diff
@@ -10,3 +10,13 @@ ssize_t __weak z_impl_hwinfo_get_device_id(u8_t *buffer, size_t length)
 {
 	return -ENOTSUP;
 }
+
```

Comment:
I think even the weak implementation can provide some greater functionality, although it would also make sense if that were done in a different PR.

For example, if a device does not support a reset-reason register, then a few `uint32_t`s or something along those lines could be queried for flags & magic values. Just so that it isn't zero'ed with the rest of BSS it would need to go into a separate linker section for persistent data.

Clearly, this may have noise when a device is powered on, hence the `MAGIC` values in [reboot(2)](https://man7.org/linux/man-pages/man2/reboot.2.html). I don't know if Zephyr has a "reboot" system call already.. does it?

As long as the CRT doesn't wipe values, they can can persist between reboots, even when using DRAM if the system doesn't lose power (which is how Android handles reboot reasons, IIRC).

Someone already mentioned that the reset reason could be passed from BL to Zephyr with some notion of "command line arguments" or something similar, but that's also out of scope here.

Reply:
I would say something like that even has value when the arch does have a reset-reason register, because at least everything would be kind of forced to fit into a vendor-neutral model.
My 2c though.

Definitely an extremely useful feature!